### PR TITLE
fix: handle text input in react native generator

### PR DESCRIPTION
**Pull request by Othman2001**
## Description
adding text input in the react native generator, react native text input handling is missed, we should convert input tag from react to TextInput and add styles, onChange and so on

this is a fix for this issue:
https://github.com/BuilderIO/mitosis/issues/1390

### CODE REVIEW

**Review by ShehabThndr — APPROVED** [review]

File: packages/core/src/generators/react-native/index.ts [hunk 1]
```diff
@@ -126,13 +126,15 @@ const PROCESS_REACT_NATIVE_PLUGIN: Plugin = () => ({
             if (node.bindings.onClick) {
               node.name = 'Pressable';
             }
+            if(node.bindings.onChange && node.bindings.value){
+              node.name = 'TextInput'
+            }
```
[the remark this answers is not in the document]
Comment (Othman2001) [reply packages/core/src/generators/react-native/index.ts:131]: @samijaber  yeah, makes sense i do it 👍

File: packages/core/src/generators/react-native/index.ts [hunk 2]
```diff
@@ -120,19 +120,20 @@ const PROCESS_REACT_NATIVE_PLUGIN: Plugin = () => ({
         if (isMitosisNode(node)) {
           // TODO: handle TextInput, Image, etc
           if (isChildren({ node })) {
-            node.name = '';
```
Comment (samijaber) [inline packages/core/src/generators/react-native/index.ts:123]: let's keep this logic here, removing it is unrelated to this PR

Comment (Othman2001) [reply packages/core/src/generators/react-native/index.ts:123]: @samijaber yeah makes sense, I have reverted this line

Comment (Othman2001) [reply packages/core/src/generators/react-native/index.ts:123]: I noticed that tests fails in marko.test.ts and solit.ts, i will check why this happened

Comment (Othman2001) [reply packages/core/src/generators/react-native/index.ts:123]: @samijaber  I managed to fix the failed test by updating snapshots and re-running the tests locally to check on everything.

**Review by samijaber — APPROVED** [review]
Thanks for the contribution!

**Pull request merged**